lib/shopify_api/api_access.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module ShopifyAPI
+  class ApiAccess
+    SCOPE_DELIMITER = ','
+
+    def initialize(scope_names)
+      if scope_names.is_a?(String)
+        scope_names = scope_names.to_s.split(SCOPE_DELIMITER)
+      end
+
+      store_scopes(scope_names)
+    end
+
+    def covers?(scopes)
+      scopes.compressed_scopes <= expanded_scopes
+    end
+
+    def to_s
+      to_a.join(SCOPE_DELIMITER)
+    end
+
+    def to_a
+      compressed_scopes.to_a
+    end
+
+    def ==(other)
+      other.class == self.class &&
+        compressed_scopes == other.compressed_scopes
+    end
+
+    alias :eql? :==
+
+    def hash
+      compressed_scopes.hash
+    end
+
+    protected
+
+    attr_reader :compressed_scopes, :expanded_scopes
```

Comment:
Do we need any tests that ensure `store_scopes` creates these two attributes as expected?

Reply:
We don't  because the behaviour `store_scopes` is a private method, while the `compressed_scopes` and `expanded_scopes` are protected. These will only be used within the context of the object itself to allow it to perform common operations that are public.

Reply:
Ah, I missed the `protected` line above!